crates/wasmtime/src/runtime/gc/enabled/rooting.rs

```diff
@@ -925,6 +943,15 @@ impl<T: GcRef> Rooted<T> {
         gc_ref.hash(state);
         Ok(())
     }
+
+    /// Cast `self` to a `Rooted<U>`.
+    ///
+    /// It is the caller's responsibility to ensure that `self` is actually a
+    /// `U`. Failure to uphold this invariant will be memory safe but will
+    /// result in general incorrectness such as panics and wrong results.
+    pub(crate) fn cast<U: GcRef>(self) -> Rooted<U> {
```

Comment:
Should this (and below) perhaps be called `unchecked_cast` to encourage more scrutiny at callsites?